sandcastle/api.py

```diff
@@ -502,26 +502,45 @@ def _do_exec(
                 continue
         raise SandcastleException("Unable to connect to the kubernetes API server.")
 
-    def _prepare_mdir_exec(
-        self, command: List[str], target_dir: Optional[Path] = None
+    def _prepare_exec(
+        self,
+        command: List[str],
+        target_dir: Optional[Path] = None,
+        env: Optional[Dict] = None,
+        cwd: Union[str, Path] = None,
     ) -> Tuple[Path, Path]:
         """
-        wrap a command to exec in the sandbox in a script
+        wrap a command for exec into a script file and place it to sandbox
 
         :param command: command to wrap
         :param target_dir: a dir in the sandbox where the script is supposed to be copied
+        :param env: env vars to set in the script.sh so they are available to the "command"
+        :param cwd: run the command in this directory, defaults to a mapped dir
+               or a temporary directory if mapped_dir is not set
         :return: (path to the sync'd dir within sandbox, path to the script within sandbox)
         """
         cmd_str = " ".join(f"{shlex.quote(c)}" for c in command)
 
-        root_target_dir = Path(target_dir or self._do_exec(["mktemp"]).strip())
+        root_target_dir = Path(target_dir or self._do_exec(["mktemp", "-d"]).strip())
         # this is where the content of mapped_dir will be
         unique_dir = root_target_dir / self.pod_name
         script_name = "script.sh"
 
-        # git checkout - oc cp does not preserve the file mode and makes the repo dirty
         # set -e - fail the script when a command fails
-        script_template = f"#!/bin/bash\nset -e\ncd {unique_dir}\nexec {cmd_str}\n"
+        script_template = (
+            "#!/bin/bash\n"
+            "set -e\n"
```

Comment:
```suggestion
            "set -eu\n"
```

Reply:
since we are not using any vars in the generated script, `-u` is not needed - on the other hand, if the script ever evolves in future this could catch valid errors

it should also cause no harm to users since their custom inline scripts will run in a subshell